src/main/java/com/mentoring/ecommerce/application/service/product/ProductSaveService.java

```diff
@@ -0,0 +1,41 @@
+package com.mentoring.ecommerce.application.service.product;
+
+
+import com.mentoring.ecommerce.application.port.in.SaveProductUseCase;
+import com.mentoring.ecommerce.application.port.out.S3HandlerPort;
+import com.mentoring.ecommerce.application.port.out.SaveProductPort;
+import com.mentoring.ecommerce.domain.Product;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.UUID;
+
+import static org.apache.commons.codec.binary.Base64.decodeBase64;
+
+@RequiredArgsConstructor
+@Service
+public class ProductSaveService implements SaveProductUseCase {
+
+    private final S3HandlerPort uploadPort;
+    private final SaveProductPort savePort;
+    @Value("${s3.bucket-name}")
+    private String bucketName;
+
+    @Override
+    public Product saveProduct(final Product product) {
+        String fullPath = null;
+        if (product.getPath() != null) {
+            byte[] photo = decodeBase64(product.getPath());
+            final InputStream fis = new ByteArrayInputStream(photo);
+            final String filePath = "product-photos";
+            final String fileName = String.valueOf(UUID.randomUUID());
+            uploadPort.upload(filePath, fis, fileName);
+            fullPath = bucketName.concat("/").concat(filePath).concat("/").concat(fileName);
```

Comment:
I was struggling to find which parameter I should use to save the path to the database, so I left it that way for now.
What is the best way to save images in the database?

Reply:
Will: Yes, OK. But it's important to save the file extension too, otherwise you and the S3 won't know what kind of file it is.

An interesting improvement is to create an `AmazonUtils` file to keep this kind of junk code that doesn't fall under the responsibility of the Service.